tests/core_tests/validations/user_functions.py

```diff
@@ -0,0 +1,83 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+
+from validations.base_functions import get_key_if_exists
+
+validation_memory = []
+
+
+## (user-written) validation functions
+
+## ---------------------------------------------------
+def validate_transaction_charged_amount(event, exp_event, exp_data):
+    """This function checks if an expected minimum amount of kWh has been charged.
+    For this to work, a previous call of 'set_test_memory()' has to have been made with
+    'transaction_started' as its target event.
+
+    Args:
+        event (Any): Received event
+        exp_event (optional - not needed): (always assumed to be 'transaction_finished')
+        exp_data (Any): Expected minimum amount of kWh to be charged.
+
+
+    Returns:
+        tuple(bool, str): True if valid result, else False. , result message (incl. error, if applicable)
+    """
+    global validation_memory
+
+    _, pre_event_value_from_memory = get_key_if_exists(validation_memory[0], ['transaction_started', 'energy_Wh_import'])
+    event_found, received_event_value = get_key_if_exists(event, ['transaction_finished', 'energy_Wh_import'])
+
+    if event_found == False:
+        return False, str("Error: Expected data is missing from event!")
+
+    if (float(received_event_value) - float(pre_event_value_from_memory)) > float(exp_data):
+        validation_memory = []
+        return True, str("Ok")
+
+    return False, str(f"Error: Event value ({received_event_value}) too low! "
```

Comment:
the return values here are pretty close to exceptions, maybe we can use exceptions instead?